core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala

```diff
@@ -43,6 +43,8 @@ class LocalSparkCluster(
   private val localHostname = Utils.localHostName()
   private val masterActorSystems = ArrayBuffer[ActorSystem]()
   private val workerActorSystems = ArrayBuffer[ActorSystem]()
+  // exposed for testing
+  var masterWebUIPort = -1
```

Comment:
Minor nit: we could use the `@VisibleForTesting` annotation here, but not a big deal (the codebase is a bit inconsistent about this).